### PR TITLE
fixes credhub to work with aws mysql aurora

### PR DESCRIPTION
**Pull request by churib**
Hi,

here is a fix for credhub with AWS Aurora MySQL as dbms.

Credhub with Aurora MySQL worked in prior versions of credhub with an older mariadb driver without effort.

Now the mariadb driver (at least for versions < 3) can connect to AWS Aurora MySQL, when `aurora` is used as replication parameter in the connection url, see [here](https://mariadb.com/kb/en/failover-and-high-availability-with-mariadb-connector-j-for-2x-driver/#failover-and-load-balancing-modes) and [here](https://mariadb.com/kb/en/failover-and-high-availability-with-mariadb-connector-j-for-2x-driver/#specifics-for-amazon-aurora).

This patch adds an optional job property `replication`. When set, the connection url will be like `jdbc:mariadb:<REPLICATION>://some-host:3306/some-database` for the storage type `mysql` and results in error for the storage type `postgres`.

### CODE REVIEW

**Comment by benjaminguttmann-avtq** [comment]
@peterhaochen47 @ramonskie @swalchemist @hsinn0 someone able to review this PR ? 

Thank you :)

**Comment by peterhaochen47** [comment]
This PR is the 5th item on our current work backlog, so we should get to it soon.

File: jobs/credhub/templates/application_spring.yml.erb [hunk 1]
```diff
@@ -23,12 +23,19 @@
...
+      url = "jdbc:mariadb:#{replication}://#{host}:#{port}/#{database}?autoReconnect=true"
+    else
+      url = "jdbc:mariadb://#{host}:#{port}/#{database}?autoReconnect=true"
+    end
 
     properties['spring']['flyway']['locations'] << 'classpath:/db/migration/mysql'
     properties['spring']['datasource'] = {
       'username' => username,
       'password' => password,
-      'url' => "jdbc:mariadb://#{host}:#{port}/#{database}?autoReconnect=true",
```
Comment (bruce-ricard) [inline jobs/credhub/templates/application_spring.yml.erb:31]: After investigation, it seems to us like the `autoReconnect=true` parameter from this URL is enough to connect to Aurora MySQL. 

We have noticed that if we take if out, we are not able to connect; but when we keep it, we are unable to reproduce the connection error. Without the double negative: CredHub connects and fully works with Aurora.

Please let us know if your experience is different.

**Review by bruce-ricard — CHANGES_REQUESTED** [review]
Hi, sorry for taking so long to reply.

Please see inline comment about the `autoReconnect` parameter.

TL;DR: we can't reproduce the issue this PR is trying to solve.

Thank you.

File: jobs/credhub/templates/application_spring.yml.erb (hunk 1 above)
Comment (swalchemist) [reply jobs/credhub/templates/application_spring.yml.erb:31]: @churib @benjaminguttmann-avtq ^^^ We're confused about how it's failing for you because 'autoReconnect=true' is always enabled for CredHub. 

This is with mariadb 2.7.6. The behavior is different with newer mariadb releases, which we're also investigating.

**Comment by churib** [comment]
Hi @bruce-ricard ,

sorry for replying late.

I can reproduce this issue with
- connection string of the form jdbc:mariadb://HOST:3306/credhub?autoReconnect=true ,
- Aurora with engine version 5.7.mysql_aurora.2.10.2 ,
- credhub-release version 2.12.2 ,
- so mariadb jdbc java client version 2.7.5 .

Which parameters have you tried?

**Comment by peterhaochen47** [comment]
Hello @churib, we have attempted to reproduce the error with the credhub/Aurora SQL/mariadb jdbc java client versions as well as the connection string format you mentioned, but still cannot. Could we ask you to:
- Try to test again on the latest credhub version
- Provide the complete reproduction steps, including any deploy information (steps, manifest, infrastructure info)

**Comment by bruce-ricard** [comment]
Can't reproduce. Please re-open if you can provide reproduction steps.
